pkg/stub/handler.go

```diff
@@ -278,6 +281,32 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 			if newVersion <= currVersion {
 				return nil
 			}
+			if len(srcfg.Spec.InstallType) == 0 &&
+				h.samplesResource.Status.InstallType != v1alpha1.CentosSamplesDistribution {
+				err := fmt.Errorf("cannot change install type from %q to %q",
+					h.samplesResource.Status.InstallType, srcfg.Spec.InstallType)
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
+			}
+			if len(srcfg.Spec.InstallType) != 0 &&
+				srcfg.Spec.InstallType != h.samplesResource.Status.InstallType {
+				err := fmt.Errorf("cannot change install type from %q to %q",
+					h.samplesResource.Status.InstallType, srcfg.Spec.InstallType)
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
+			}
+
+			if len(srcfg.Spec.Architectures) == 0 &&
+				(len(h.samplesResource.Status.Architectures) != 1 ||
+					h.samplesResource.Status.Architectures[0] != x86) {
+				err := fmt.Errorf("cannot change from architecture list %#v to %#v",
+					h.samplesResource.Status.Architectures, srcfg.Spec.Architectures)
+				return h.processError(srcfg, v1alpha1.SamplesUpdateFailed, err, "%v")
+			}
+			if len(srcfg.Spec.Architectures) != 0 &&
+				!reflect.DeepEqual(srcfg.Spec.Architectures, h.samplesResource.Status.Architectures) {
```

Comment:
this is an order dependent check, right?  seems fragile.

Reply:
Are you referring to the order of the conditionals within this if, or this if in relation to the prior if ?

That said, the switch to a k/v ConfigMap will reswizzle this anyway.

Reply:
i'm referring to the order of the values in the arrays.  You're comparing two arrays, I assume that compare will fail if the two arrays contain the same values in a different order, whereas from our perspective they should be considered equal.

Reply:
ah gotcha 

and yep, a stackoverflow lead me to believe it would not matter, but just tried it in go playground and order does matter

But as I said before, I think the switch to configmap will negate this.